app/code/Magento/Analytics/ReportXml/DB/ColumnsResolver.php

```diff
@@ -76,7 +76,7 @@ public function getColumns(SelectBuilder $selectBuilder, $entityConfig)
             $columnName = $this->nameResolver->getName($attributeData);
             if (isset($attributeData['function'])) {
                 $prefix = '';
-                if (isset($attributeData['distinct']) && $attributeData['distinct'] == true) {
+                if ((bool) $attributeData['distinct']) {
```

Comment:
Will throw notice when not set, `(bool)` casting is redundant here.

Correct replacement: `!empty($attributeData['distinct'])`